frontend/src/queries/nodes/DataTable/utils.ts

```diff
@@ -33,16 +33,38 @@ export function getColumnsForQuery(query: DataTableNode): HogQLExpression[] {
     return query.columns ?? getDataNodeDefaultColumns(query.source)
 }
 
-export function extractExpressionComment(query: string): string {
+/**
+ * Extracts the comment and/or alias from a query. For display only!
+ *
+ * This is a rather naive implementation using basic string operations.
+ * Its output is not cleaned for usage in queries!
+ */
+export function extractCommentOrAlias(query: string): string {
+    if (query.match(/ as (`[^`]+`|"[^"]+"|[a-zA-Z$][a-zA-Z0-9_$]*)\s*$/)) {
+        const comment = query.split(' as ').pop()?.trim() || query
+        if ((comment.startsWith('`') || comment.startsWith('"')) && comment.endsWith(comment[0])) {
+            return comment.slice(1, -1)
+        }
+        return comment
+    }
     if (query.includes('--')) {
         return query.split('--').pop()?.trim() || query
     }
     return query
 }
 
-export function removeExpressionComment(query: string): string {
+/**
+ * Removes the comment and/or alias from a query, keeping just the SQL. For display only!
+ *
+ * This is a rather naive implementation using basic string operations.
+ * Its output is not cleaned for usage in queries!
+ */
+export function removeCommentOrAlias(query: string): string {
     if (query.includes('--')) {
-        return query.split('--').slice(0, -1).join('--').trim()
+        query = query.split('--').slice(0, -1).join('--').trim()
+    }
+    if (query.match(/ as (`[^`]+`|"[^"]+"|[a-zA-Z\$][a-zA-Z0-9\_\$]*)$/)) {
```

Comment:
These two regexes only seem to differ by "_" and "$" being escaped in character sets, and whitespace being allowed at the end of the extraction one. Are these differences important, or should these be the same? I think this would be clearer if the regexes were put in file-level constants.
BTW I think the case insensitivity flag is missing, this wouldn't match the valid "AS" or "As"